src/Avalonia.Animation/Animator`1.cs

```diff
@@ -28,21 +31,20 @@ public abstract class Animator<T> : AvaloniaList<AnimatorKeyFrame>, IAnimator
         public Animator()
         {
             // Invalidate keyframes when changed.
-             this.CollectionChanged += delegate { _isVerifiedAndConverted = false; };
+            this.CollectionChanged += delegate { _isVerifiedAndConverted = false; };
         }
 
         /// <inheritdoc/>
         public virtual IDisposable Apply(Animation animation, Animatable control, IObservable<bool> match, Action onComplete)
         {
-             if (!_isVerifiedAndConverted)
+            if (!_isVerifiedAndConverted)
                 VerifyConvertKeyFrames();
 
-            return match
-                .Where(p => p)
-                .Subscribe(_ =>
-                {
-                    var timerObs = RunKeyFrames(animation, control, onComplete);
-                });
+            return match.DistinctUntilChanged()
+                        .ObserveOn(Avalonia.Threading.AvaloniaScheduler.Instance)
```

Comment:
This seems to be necessary since i'm having LinkedList exceptions on Bindings whenever the animation cancels...

Reply:
Ahh this is just delaying the inevitable; that exception still happens somehow.